### ISSUE TITLE
Unable to connect to IRC server.

### ISSUE THREAD

**Issue by Ruthalas** [evan-buss/openbooks#102]
Hello! I am experiencing the following issue when attempting to use OpenBooks.
When the webUI is loaded, it gives the error: "Unable to connect to IRC server."

Here is a chunk of the log, starting from a restart of the container:
```
2022/09/12 10:27:40 SERVER: Base Path: /
2022/09/12 10:27:40 SERVER: OpenBooks is listening on port 80
2022/09/12 10:27:40 SERVER: Download Directory: /books
2022/09/12 10:27:40 SERVER: Open http://localhost:80/ in your browser.
2022/09/12 10:27:40 SERVER: Base Path: /
2022/09/12 10:27:40 SERVER: OpenBooks is listening on port 80
2022/09/12 10:27:40 SERVER: Download Directory: /books
2022/09/12 10:27:40 SERVER: Open http://localhost:80/ in your browser.
2022/09/12 10:27:49 SERVER: Client connected from XXX.XXX.XXX.XXX:XXXX
2022/09/12 10:27:49 CLIENT (straight_contracst_1): New client created.
2022/09/12 10:27:49 CLIENT (straight_contracts_1): CONNECT Message Received
2022/09/12 10:27:50 CLIENT (straight_contracts_1): EOF
2022/09/12 10:27:50 CLIENT (straight_contracts_1): EOF
```

In my docker configuration I am pulling evanbuss/openbooks:latest, so I believe I am up to date.
(I don't see a version number in the webUI anywhere.)
I have tried setting the name flag, but it did not change the outcome.

Any assistance would be appreciated!

If there is any other information I can provide, please let me know!

**Comment by unai-ndz** [comment]
Are you using a reverse proxy? This happens to me using one but not if I connect directly.

**Comment by mtnezm** [comment]
Can confirm this on a Kubernetes setup too (Docker based). It seems work fine as soon as `hostNetwork: true` is added to the pod configuration.

**Comment by AlexMuresan** [comment]
Having the same issue as well. I'm guessing there's a problem with the IRC.

**Comment by wisewtf** [comment]
Same thing over here. Comes and goes infrequently. 

I only get EOF when it fails to connect.

**Comment by evan-buss** [comment]
I am unable to reproduce this issue on my end. To anyone experiencing this issue, please run OpenBooks with the `--log` flag so that IRC logs are saved to a file for each client connection.

Docker Compose:

```
command: --persist --log
```

**Comment by Ruthalas** [comment]
I opened the application to test this after enabling logging and had no issue connecting.
I removed the logging and still had no issue connecting.

Possibly a temporary issue on the IRC side?
Regardless, appreciate the effort.

**Comment by ncryptedV1** [comment]
Issue exists when keeping TLS authentication enabled, as IRCHighway certs seem to be self-signed. With their root CA added to the cert store or with `--tls=false` the connection works. Despite that, no commands seem to reach the server. User and messages also can't manually be found in the users list/chat log of the channel when connecting with a common IRC client.

**Comment by evan-buss** [comment]
> Issue exists when keeping TLS authentication enabled, as IRCHighway certs seem to be self-signed. With their root CA added to the cert store or with `--tls=false` the connection works.

TLS connections don't validate the issuer so this shouldn't be the issue.

https://github.com/evan-buss/openbooks/blob/91bb21ac7d8d2b09e1117d1a573184b000feadb0/irc/irc.go#LL32C2-L32C2

**Comment by ncryptedV1** [comment]
> TLS connections don't validate the issuer so this shouldn't be the issue.
> 
> https://github.com/evan-buss/openbooks/blob/91bb21ac7d8d2b09e1117d1a573184b000feadb0/irc/irc.go#LL32C2-L32C2
Agreed, turned out to be a network configuration problem. Works flawlessly now.

**Comment by pjannink** [comment]
HI, I also have same issue. How did you resolve the network issue? I use local port 8081 / container port 80 and forwarded port 8081 in my router to my synology NAS where I am running Docker and the Openbooks-container. So a bit clueless why it says "not connected".....

**Comment by MMilosz** [comment]
Running `docker run -d -p 8080:80 -v ~/Downloads:/books evanbuss/openbooks --persist` also gave me this kind of error but having changed `8080:80` to `80:80` I can finally establish connection. No idea if this workaround works for everyone

**Comment by pjannink** [comment]
Does not work for me. See my setup stated above. Openbooks should create a default user according evan-buss but no signs of a userid in the low left corner of the Openbooks-webpage: it states "Not connected" there.

**Comment by takov751** [comment]
I have solved it with simply adding one bit next to `presist`

`-s "irc.irchighway.net:9999"`
with this I am able to connect to irchightway.net without issue.

**Comment by pjannink** [comment]
Sadly not here. I'm working with Docker GUI on Synology and Portainer, but with Docker's export/import function I could edit the cmd line in the json file and recreated the container. Edited line in the JSON:

   "cmd" : "./openbooks server --dir /books --port 80 --persist -s irc.irchighway.net:9999",

According Portainer the settings are now:

CMD | ./openbooks server --dir /books --port 80 --persist -s irc.irchighway.net:9999
ENTRYPOINT | ./openbooks server --dir /books --port 80

But...still not working...Now idea how I could edit the Entrypoint...
Hope someone can help me out..

**Comment by zenjubbers** [comment]
For those of you using nginx proxy manager, make sure you enable "websocket support" for openbooks. I was seeing an unable to connect error and the lower left of the page would show disconnected until I enabled websocket support.

**Comment by ljo123** [comment]
> For those of you using nginx proxy manager, make sure you enable "websocket support" for openbooks. I was seeing an unable to connect error and the lower left of the page would show disconnected until I enabled websocket support.

How are you using openbooks with NPM? Both need to listen on docker internal port 80.

**Comment by zenjubbers** [comment]
> How are you using openbooks with NPM? Both need to listen on docker internal port 80.

Just map to an unused port on your docker then in NPM, You'd forward traffic to the docker server on said port.

For example, I use port 8023 on the docker server for the openbook container.
    ports:
      - 8023:80

Then in my NPM, I simply have it forward traffic to http://<IP of docker server>:8023

**Comment by ljo123** [comment]
> > How are you using openbooks with NPM? Both need to listen on docker internal port 80.
> 
> Just map to an unused port on your docker then in NPM, You'd forward traffic to the docker server on said port.
> 
> For example, I use port 8023 on the docker server for the openbook container. ports: - 8023:80
> 
> Then in my NPM, I simply have it forward traffic to http://{IP of docker server}:8023

Nice thanks. I was confused and thought that you could only use the internal port once like you do with external. Docker is smarter than I thought!

**Comment by IMBeniamin** [comment]
-tls=false got me a connection, but commands are not forwarded, so I was able to reproduce 

> Issue exists when keeping TLS authentication enabled, as IRCHighway certs seem to be self-signed. With their root CA added to the cert store or with `--tls=false` the connection works. **Despite that**, no commands seem to reach the server. User and messages also can't manually be found in the users list/chat log of the channel when connecting with a common IRC client.

I have also tried using 

> I have solved it with simply adding one bit next to `presist`
> 
> `-s "irc.irchighway.net:9999"` with this I am able to connect to irchightway.net without issue. so basically i just changed default irc port

but had no success

**Comment by peseotni** [comment]
in my case it has to do with nginx proxy manager. If I put in the ip address, it works, it connects, searches, and downloads, but it doesn't work through the proxy

**Comment by conortimmsy** [comment]
> For those of you using nginx proxy manager, make sure you enable "websocket support" for openbooks. I was seeing an unable to connect error and the lower left of the page would show disconnected until I enabled websocket support.

Thanks, man. Just what I was missing!!

**Comment by ShadowsDieThrice** [comment]
Issue still persists. Not using nginx proxy manager, but the connection still drops out sporadically. Have tried running both dockerised and directly from the exe with similar results.

**Comment by roadkingvrod** [comment]
I needed to add -s irc.irchighway.net:9999 to my "command" options in docker compose.  seems irchighway changed their port.

**Comment by tfig-dev** [comment]
still an issue, right? I've tried so many ways and just worken once. this is my docker.compose.yml
`
  openbooks:
    container_name: openbooks
    command: --persist --name my_user_name -s irc.irchighway.net:6697
    image: evanbuss/openbooks:latest
    volumes:
      - /openbooks:/books
    ports:
      - 8875:80`
      
      I just had checked the both SSL IRC ports 6697 / 9999 and nothing.
      the NPM redirection works
